homeassistant/util/dt.py

```diff
@@ -4,17 +4,18 @@
 from contextlib import suppress
 import datetime as dt
 import re
-from typing import Any
-
-try:
-    import zoneinfo
-except ImportError:
-    from backports import zoneinfo
+import sys
+from typing import Any, cast
 
 import ciso8601
 
 from homeassistant.const import MATCH_ALL
 
+if sys.version_info[:2] >= (3, 9):
+    import zoneinfo  # pylint: disable=import-error
```

Comment:
@cdce8p if you want to improve pylint a little bit more.

Mypy understands blocks inside `sys.version_info` but pylint doesn't. But on the other side pylint doesn't complain about
```python
try:
  import x
except ImportError
  import y
```
while mypy has this issue.

It'd be nice to support `sys.version_info` in pylint as well.

Reply:
You could use the `ignored-modules` option, [documentation](http://pylint.pycqa.org/en/latest/technical_reference/features.html#typecheck-checker-options) or wait for https://github.com/PyCQA/pylint/pull/4468 😃